ckan-docker-base.sh

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+
+set -e
+
+
+set_vars() {
+    local ckan_version_ref="$1"
+    local env="$2"
+
+    ckan_version=$(cat "ckan-$ckan_version_ref/VERSION.txt")
+    python_version=$(cat "ckan-$ckan_version_ref/PYTHON_VERSION.txt")
+    python_dockerfile=Dockerfile.py$python_version
+    tag_name="ckan/ckan-$env:$ckan_version"
+    python_tag_name="ckan/ckan-$env:$ckan_version-py$python_version"
+    if [ "$ckan_version" = "master" ]; then
+        ckan_tag=$ckan_version
+        alt_tag_name=$tag_name
+        python_alt_tag_name=$python_tag_name
+    else
+        ckan_tag=ckan-$ckan_version
+        ckan_version_major=$(echo "$ckan_version" | cut -d'.' -f1)
+        ckan_version_minor=$(echo "$ckan_version" | cut -d'.' -f2)
+        alt_tag_name="ckan/ckan-$env:$ckan_version_major.$ckan_version_minor"
+        python_alt_tag_name="ckan/ckan-$env:$ckan_version_major.$ckan_version_minor-py$python_version"
+    fi
+
+}
+
+
+build_images() {
+    local ckan_version_ref="$1"
+    local env="$2"
+
+    set_vars "$ckan_version_ref" "$env"
+
+    if [ -f "ckan-$ckan_version_ref/$python_dockerfile" ]; then
+        # Build Python image if there's a separate .pyXX Dockerfile
+        DOCKER_BUILDKIT=1 docker build \
+            --build-arg="ENV=$env" \
+            --build-arg="CKAN_REF=$ckan_tag" \
+            -t "$python_tag_name" \
+            -t "$python_alt_tag_name" \
+            -f "ckan-$ckan_version_ref/$python_dockerfile" \
+            "ckan-$ckan_version_ref"
+    fi
+
+    # Build image 
+    DOCKER_BUILDKIT=1 docker build \
+        --build-arg="ENV=$env" \
+        --build-arg="CKAN_REF=$ckan_tag" \
+         -t "$tag_name" \
+         -t "$alt_tag_name" \
+         -t "$python_tag_name" \
+         -t "$python_alt_tag_name" \
+         "ckan-$ckan_version_ref"
+}
+
+
+push_images() {
+    local ckan_version_ref="$1"
+    local env="$2"
+
+    set_vars "$ckan_version_ref" "$env"
+    
+    echo "Pushing $tag_name"
+    docker push "$tag_name"
+    echo "Pushing $alt_tag_name"
+    docker push "$alt_tag_name"
+    echo "Pushing $python_tag_name"
+    docker push "$python_tag_name"
+    echo "Pushing $python_alt_tag_name"
+    docker push "$python_alt_tag_name"
+
+}
+
+
+show_versions() {
+    echo "Versions built currently:"
+    cat ckan-*/VERSION.txt | sort -t. -k1,1n -k2,2n | sed 's/^/* /'
+}
+
+
```

Comment:
check_ckan_version_ref() {
    local ckan_version_ref="$1"
    # If input is in the format X.Y.Z (e.g., "2.11.0"), trim to X.Y (e.g., "2.11")
    if [[ "$ckan_version_ref" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
        ckan_version_ref="${ckan_version_ref%.*}"
    fi
    echo "$ckan_version_ref"
}